lib/SILGen/SILGenStmt.cpp

```diff
@@ -74,7 +74,9 @@ void SILGenFunction::eraseBasicBlock(SILBasicBlock *block) {
   assert(block->pred_empty() && "erasing block with predecessors");
   assert(block->empty() && "erasing block with content");
   if (block == StartOfPostmatter) {
-    StartOfPostmatter = &*std::next(block->getIterator());
+    StartOfPostmatter = (std::next(block->getIterator()) == F.end())
+                            ? nullptr
+                            : &*std::next(block->getIterator());
```

Comment:
So this isn't quite as useful as I would hope. The problem is that the `StartOfPostmatter` case requires the dereference (`&*std::next`), while the `StrongReleaseInst` case does not. This means that the `&*` can't be factored into the function, and has to occur on the return value. Which means you still have to check if the return value of `next_or_null` is `null`, because you can't call `&*` on a nullptr.